pkg/auth/server/login/login.go

```diff
@@ -164,19 +165,26 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 		failed(errorCodeUserRequired, w, req)
 		return
 	}
+	var result string
+	defer func() {
+		metrics.Record(result)
+	}()
 	user, ok, err := l.auth.AuthenticatePassword(username, password)
 	if err != nil {
 		glog.Errorf(`Error authenticating %q with provider %q: %v`, username, l.provider, err)
 		failed(errorpage.AuthenticationErrorCode(err), w, req)
+		result = "error"
```

Comment:
I 'd prefer the result be a constant

Reply:
Generally we don't create constants for single use variables.